enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,336 @@
+  - "@mrunalp"
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/OCPNODE-1628
+see-also:
+  - "/enhancements/this-other-neat-thing.md"
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# Add CRD ImagePolicy to config.openshift.io/v1alpha1
+
+## Summary
+
+This enhancement propose an ImagePolicy CRD to provide a way to set image signature verification configuration on Openshift.
+
+## Motivation
+
+### User Stories
+
+As an OpenShift user, I want to verify the container images signed using sigstore
+tools, so that I can utilize the increased security of my software supply chain.
+
+The user can create an ImagePolicy instance specifying the images/repositories to be verified and their policy. The MCO will write the configuration for signature verification. Once this is done, CRI-O can verify the images/repositories.
+
+### Goals
+
+- MCO container runtime config controller watches ImagePolicy instance in different namespaces and merges the instances for each namespace into a single containers-policy.json in a predefined /path/to/policies/\<NAMESPACE\>.json.
+
+- CRI-O can verify Cosign signature signed images using configuration from ImagePolicy
+
+### Non-Goals
+
+ImagePolicy CRD does not implements cosign with custom Fulcio and Rekor instance.
+
+## Proposal
+
+### Workflow Description
+
+**cluster administrator** is a human user responsible for adding signature verification configurations in
+cluster scope.
+1. The cluster administrator creates ImagePolicy yaml file with `namespace: opesnhift-config`.
+2. The cluster administrator creates ImagePolicy instance using `oc create -f imagepolicy.yaml`.
+3. The cluster administrator can get the merged cluster scope policies from `status` of ImagePolicy instances. 
+4. The cluster administrator can delete the signature verification configuration by deleting its ImagePolicy instances. 
+
+**application administrator** is a human user responsible for
+adding signature verification configurations for specific namespace.
+
+1. The application administrator creates ImagePolicy yaml file for namespace that is not `namespace: opesnhift-config`
+2. The application administrator creates ImagePolicy instance using `oc create -f imagepolicy.yaml`.
+3. The application administrator can get the merged cluster policy and policy for certain namespace from `status` of ImagePolicy instances. 
+4. The application administrator can delete the signature verification configuration by deleting its ImagePolicy instances. 
+
+### API Extensions
+
+#### Type definitions
+
+```go
+type ImagePolicySpec struct {
+	// images holds images/repositories to be verified
```

Comment:
```suggestion
	// Images holds images/repositories to be verified
```

Reply:
Checked some apis from https://github.com/openshift/api/tree/master/config/v1 This comment will be in the generated CRD schemas description, we should use the name defined in the json tag.

Reply:
Sounds good, thank you for the clarification.